Log Error when Exception occurs during start/configure (Issue #95)

alexxcons opened the pull request:
In order to Make Configuration errors at least more visible

I did not find out whats the consequence of returning "false" in `start`, seems to make no difference.

I suppose Fesa should check `d_configure_exception_message` and stop/abort the flowgraph when it is filled.

Fixes https://github.com/fair-acc/gr-digitizers/issues/95

wirew0rm commented on lib/digitizer_block_impl.cc:
```diff
@@ -700,9 +700,19 @@ namespace gr {
        d_configure_exception_message = ex.what();
+       GR_LOG_ERROR(d_logger, d_configure_exception_message );
+
+       // It looks like the block is run by gnuradio, no matter if true or false is returned here
```
I also couldn't find any use of `start()`that used the return value.

wirew0rm commented on lib/digitizer_block_impl.cc:
```diff
@@ -700,9 +700,19 @@ namespace gr {
        d_configure_exception_message = ex.what();
+       GR_LOG_ERROR(d_logger, d_configure_exception_message );
+
+       // It looks like the block is run by gnuradio, no matter if true or false is returned here
```
I also couldn't find a single use of `start()` where the result was used (or something other than true was returned) in the gnuradio source.

RalphSteinhagen replied:
As per the latest [documentation](https://www.gnuradio.org/doc/doxygen-v3.9.4.0/classgr_1_1block.html#a7f58745d1374b30a7b866406dc97850f) the return value of `gr::block.start()` doesn't seem to have a pre-sprecified value and isn't further documented.

I also reckon that the original meaning got lost in the ages. 

Maybe the GnuRadio oracle @marcusmueller can enlighten us ...

alexxcons replied:
Resolving the conversation for now ... if we can find out more about it later on, lets open a new MR/issue.

wirew0rm commented on lib/digitizer_block_impl.cc:
```diff
@@ -700,9 +700,19 @@ namespace gr {
+       GR_LOG_ERROR(d_logger, d_configure_exception_message );
+
+       // It looks like the block is run by gnuradio, no matter if true or false is returned here
+       // TODO: "start" is called by gnuradio itself. Check if it is possible to stop the block from running when false is returned here.
```
Gnuradio's own modules all return true, but the `tcp_sink` block just throws, so maybe gnuradio will actually catch the exception by itself and stop flowgraph execution and we can just rethrow the exception? Could also be an oversight in the gr_networking module because I haven't seen it anywhere else.
https://github.com/gnuradio/gnuradio/blob/7712360ddbe262ed6a48d622a695aaaaa4993795/gr-network/lib/tcp_sink_impl.cc#L76

wirew0rm reviewed (approved):
Looks good to me :+1:

RalphSteinhagen reviewed (approved):
Also OK from my side. :+1:

alexxcons commented:
I did some more experimenting on it. Meanwhile I think best thing to do is to just call `stop` on the block, so that it stops running:

https://github.com/fair-acc/gr-digitizers/pull/96/commits/b01fb17c1d4a0dd8c7aa744907c14f24d90dbe5d

Seems to work fine (The result will be that FESA starts to complain that no data is received after few seconds)

Added the comment only to the first return, to dont duplicate it, and modified it to tell what I found out so far.

wirew0rm reviewed (approved):
Looks good!

alexxcons commented:
Merged, thank you for review !
